helix-term/src/handlers/signature_help.rs

```diff
@@ -0,0 +1,345 @@
+use std::sync::Arc;
+use std::time::Duration;
+
+use helix_core::syntax::LanguageServerFeature;
+use helix_event::{
+    canceable_future, cancelation, register_hook, send_blocking, CancelRx, CancelTx,
+};
+use helix_lsp::lsp;
+use helix_view::document::Mode;
+use helix_view::events::{DocumentDidChange, SelectionDidChange};
+use helix_view::handlers::lsp::{SignatureHelpEvent, SignatureHelpInvoked};
+use helix_view::Editor;
+use tokio::sync::mpsc::Sender;
+use tokio::time::Instant;
+
+use crate::commands::Open;
+use crate::compositor::Compositor;
+use crate::events::{OnModeSwitch, PostInsertChar};
+use crate::handlers::{rope_ends_with, Handlers};
+use crate::ui::lsp::SignatureHelp;
+use crate::ui::Popup;
+use crate::{job, ui};
+
+#[derive(Debug)]
+enum State {
+    Open,
+    Closed,
+    Pending { request: CancelTx },
+}
+
+/// debounce timeout in ms, value taken from VSCode
+/// TODO: make this configurable?
+const TIMEOUT: u64 = 120;
```

Comment:
yes please, it's significantly slower now.

Reply:
I am not really sure we want to keep adding more config options for various timeouts. It feels like a bit much and in this case in particular there is a very good reason for a larger timeout. Everytime you do anything in insert mode (change the document, change the selection,...) helix will request signature help information from the server. This is necessary (and also how vscode handles this) but can lead to a LOT of requests so a reasonably large debounce timeout is important. 

VSCode event uses this timeout and they usually tend to favour fast responses much more than we do (like completion requests being sent instantly to the servers)

Reply:
I too think that it's beneficial to make this configurable (shouldn't introduce much code should it?). I detest to have to wait (also this short time) for this stuff, when my machine can easily handle it. When the amount of configurable timeouts gets "too" much, maybe it makes sense to introduce a separate toml table for timeouts?

Reply:
> VSCode event uses this timeout and they usually tend to favour fast responses much more than we do (like completion requests being sent instantly to the servers)

I think completion was the only thinkg like this - when I used vscode a few years ago and watched development due to the enormous latency in the vim plugin, they were at the stage where they were adding debouncing everywhere due to tragic performance.

vscode is usable now, but when you switch to a really fast editor with low latency, it's like magic - everything happens before you even think about it :). I really like the feel of it compared to wading through mud because of a few watts.

maybe we need two kinds of idle timeouts - fast "interactive idle" for debouncing things one is actively waiting for and "background idle".